components/RefreshRemoteUser.coffee

```diff
@@ -0,0 +1,66 @@
+noflo = require 'noflo'
+
+exports.getComponent = ->
+  c = new noflo.Component
+  c.inPorts.add 'in',
+    datatype: 'object'
+  c.outPorts.add 'pass',
+    datatype: 'object'
+  c.outPorts.add 'updated',
+    datatype: 'object'
+  c.outPorts.add 'invalid',
+    datatype: 'object'
+  c.outPorts.add 'error',
+    datatype: 'object'
+
+  noflo.helpers.WirePattern c,
+    in: 'in'
+    out: ['pass', 'updated', 'invalid']
+    async: true
+  , (data, groups, out, callback) ->
+    unless data?['grid-token']
+      # If user is not logged in, there is nothing to do
+      out.pass.send data
+      do callback
+      return
+
+    unless navigator.onLine
+      # When offline we can't refresh
+      out.pass.send data
```

Comment:
This component doesn't really care whether logged in or not. The actual thing the component does should be pretty well reflected in the ports:

* User information was fine as-is (whether logged in or not)
* User information was invalid (non-working OAuth token etc)
* User information was updated from IdP